examples/mooseherder/example_1_simple_monoblock.py

```diff
@@ -0,0 +1,60 @@
+"""
+SLEDO MooseHerder/Proteus Simple Monoblock Example
+
+Example input file for running Bayesian optimisation of a simplified divertor
+monoblock using MooseHerder as the model instantiation tool and Proteus as the
+simulation tool.
+
+(c) Copyright UKAEA 2023-2024.
+"""
+
+from pathlib import Path
+from ray import tune
+from ray.tune.search.ax import AxSearch
+
+from sledo.optimiser import Optimiser
+from examples.mooseherder.design_evaluator import DesignEvaluator
+
+MOOSE_OPT = "proteus-opt"
+EXAMPLES_DIR = Path(__file__).parent.absolute()
+BASE_INPUT_FILE = EXAMPLES_DIR / "input_files" / "simple_monoblock.i"
+WORKING_DIR = EXAMPLES_DIR / "results" / "simple_monoblock"
+
+if __name__ == '__main__':
+
+    # Instantiate moose herder class.
+    design_evaluator = DesignEvaluator(
+        MOOSE_OPT,
+        BASE_INPUT_FILE,
+        WORKING_DIR,
+    )
+
+    # Define design evaluation function.
+    def evaluate(parameters):
+        data = design_evaluator.evaluate_design(parameters)
+        stress = float(data[-1][-1])
+        return {"stress": stress}
+
+    # Define a search space according to the Ray Tune API.
+    # Documentation here:
+    # https://docs.ray.io/en/latest/tune/api/search_space.html
+    search_space = {
+        "monoBArmHeight": tune.uniform(1e-3, 20e-3),
+        "monoBThick": tune.uniform(0.5e-3, 9e-3),
+    }
+
+    # Instantiate SLEDO optimiser.
+    opt = Optimiser(
+        "simple-monoblock-optimiser",
+        evaluate,
+        search_space,
+        "stress",
+        AxSearch(),
```

Comment:
Ideally, we don't want the user to have to go searching through the tune documentation to find a search algorithm, but we also don't want to limit the search algorithms available to the user unnecessarily. It might be worth setting AxSearch or some other as a sensible default. Open to suggestions on the best way to approach this.

The search algorithm object is also where input and output constraints can be encoded, which is something we want the user to be able to specify. These can be handled in `Optimiser.__init__` or we could ask the user to specify these when instantiating their search algorithm. My preference is for the former as it reduces the ray tune API knowledge required of the user.